src/core/client_channel/client_channel.cc

```diff
@@ -167,6 +168,10 @@ class ClientChannel::SubchannelWrapper
   void CancelDataWatcher(DataWatcherInterface* watcher) override
       ABSL_EXCLUSIVE_LOCKS_REQUIRED(*client_channel_->work_serializer_);
   void ThrottleKeepaliveTime(int new_keepalive_time);
+  std::string address_string() const override {
+    return grpc_sockaddr_to_string(&subchannel_->address(), true)
```

Comment:
Instead of having to duplicate this code in both client channel implementations, I suggest just changing the [`Subchannel::address()` method](https://github.com/grpc/grpc/blob/33edf1a9a3c676484333cd1cb980fcef0bc32eee/src/core/client_channel/subchannel.h#L218) to return the address as a URI string, so that this code can be a simple wrapper.  Note that there is only [one existing caller of that method](https://github.com/grpc/grpc/blob/33edf1a9a3c676484333cd1cb980fcef0bc32eee/src/core/load_balancing/health_check_client.cc#L170), which itself converts the address to URI form.

Reply:
done